README.md

```diff
@@ -187,6 +191,433 @@ class PaymentScreen extends React.Component {
+
+#### Error Code Usage
+
+The SDK exports an `ErrorCode` enum for type-safe error handling:
+
+```typescript
+import { ErrorCode } from '@stripe/stripe-terminal-react-native';
+
+// Recommended: Use ErrorCode enum for type safety
+if (error.code === ErrorCode.BLUETOOTH_ERROR) {
+  // TypeScript knows this is a valid error code
+  // IDE provides autocomplete and refactoring support
+}
+
+// Also supported: String comparison (backward compatible)
+if (error.code === 'BLUETOOTH_ERROR') {
+  // Works, but no compile-time validation
+}
+
+// Available error codes include:
+// ErrorCode.BLUETOOTH_ERROR
+// ErrorCode.READER_BUSY  
+// ErrorCode.CANCELED
+// ErrorCode.DECLINED_BY_STRIPE_API
+// ErrorCode.CARD_READ_TIMED_OUT
+// ErrorCode.REQUEST_TIMED_OUT
+// ... and 130+ additional specific codes
+```
+
+### Platform Error Conversion
+
+The React Native SDK acts as a bridge between the native Android and iOS Terminal SDKs and your React Native application. When the underlying native Terminal SDKs throw exceptions or errors, the React Native SDK automatically catches and converts them into standardized `StripeError` objects.
+
+#### Architecture Overview
+
+```text
+┌─────────────────────┐    ┌─────────────────────┐
+│   Android Terminal  │    │    iOS Terminal     │
+│        SDK          │    │        SDK          │
+└──────────┬──────────┘    └──────────┬──────────┘
+           │                          │
+           │ Platform-specific        │ Platform-specific
+           │     Errors               │     Errors
+           │                          │
+           ▼                          ▼
+┌──────────────────────────────────────────────────┐
+│           React Native SDK Bridge                │
+│        (Automatic Error Conversion)              │
+└──────────────────┬───────────────────────────────┘
+                   │
+                   │ Standardized StripeError
+                   │
+                   ▼
+┌──────────────────────────────────────────────────┐
+│          Your React Native App                   │
+└──────────────────────────────────────────────────┘
+```
+
+The SDK automatically handles different native error structures between Android and iOS, converting them to `StripeError` format:
+
+#### What You Receive as a Developer
+
+Regardless of the underlying platform, your React Native code always receives the same `StripeError` structure:
+
+```typescript
+// Consistent StripeError format across all platforms
+{
+  name: "StripeError",
+  code: "BLUETOOTH_ERROR",           // Standardized error code
+  message: "Bluetooth connection failed",
+  nativeErrorCode: "1001",          // Platform-specific code for debugging
+  metadata: {                       // Rich debugging context - see platform differences below
+    // Platform-specific debugging information
+    // Different fields available on Android vs iOS
+  },
+  paymentIntent?: { /* Related PaymentIntent */ },
+  setupIntent?: { /* Related SetupIntent */ }
+}
+```
```

Comment:
since RN is supposed to abstract this information anyways we might as well skip this in the readme too